src/components/Modal/Modal.tsx

```diff
@@ -1,27 +1,43 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { ModalOverlay } from "./ModalOverlay";
 
 export interface ModalProps {
   children: React.ReactNode;
   name: string;
-  isOpen: boolean; // Added prop
-  onClose?: () => void; // Added prop
+  isOpen: boolean;
+  onClose?: () => void;
 }
 
 export const Modal = ({ children, name, isOpen, onClose }: ModalProps) => {
```

Comment:
As a bonus/stretch goal, it'd be nice if when the user hit 'escape' on their keyboard it would also close the modal.

Reply:
Created this functionality inside of Modal.tsx and Modal.stories.tsx